src/utils/scroll.ts

```diff
@@ -10,17 +10,35 @@ const DEFAULT_THRESHOLD = 1000; // pixels
 
 export function scrollToLocation(parentEl: HTMLElement, referenceEl: HTMLElement, options: ScrollOptions = {}): void {
     const { offsets = DEFAULT_OFFSETS, threshold = DEFAULT_THRESHOLD } = options;
-    const { x: offsetX, y: offsetY } = offsets;
+    const { x: offsetXPercentage, y: offsetYPercentage } = offsets;
+    // Get the bounding client rects so that the offsetLeft of the reference element can be calculated --
+    // even if it has been transformed via rotation
+    const { left: parentLeft, top: parentTop } = parentEl.getBoundingClientRect();
+    const {
+        height: referenceHeight,
+        left: referenceLeft,
+        top: referenceTop,
+        width: referenceWidth,
+    } = referenceEl.getBoundingClientRect();
 
     const canSmoothScroll = 'scrollBehavior' in parentEl.style;
+    // Get the midpoint of the scrollable area (parent element)
     const parentCenterX = Math.round(parentEl.clientWidth / 2);
     const parentCenterY = Math.round(parentEl.clientHeight / 2);
-    const offsetCenterX = Math.round(referenceEl.clientWidth * (offsetX / 100));
-    const offsetCenterY = Math.round(referenceEl.clientHeight * (offsetY / 100));
-    const offsetScrollLeft = referenceEl.offsetLeft - parentCenterX + offsetCenterX;
-    const offsetScrollTop = referenceEl.offsetTop - parentCenterY + offsetCenterY;
-    const scrollLeft = Math.max(0, Math.min(offsetScrollLeft, parentEl.scrollWidth));
-    const scrollTop = Math.max(0, Math.min(offsetScrollTop, parentEl.scrollHeight));
+    // Get any specified offset relative to the reference element
+    const offsetX = Math.round(referenceWidth * (offsetXPercentage / 100)); // offsetX is assumed to be a percentage
+    const offsetY = Math.round(referenceHeight * (offsetYPercentage / 100)); // offsetY is assumed to be a percentage
+    // Get the offsets of the reference element relative to the viewport
```

Comment:
The use of "offset" here is a little confusing since `offsetX` and `offsetY` above mean something different. It's also relative to the parent _and_ the viewport.